server/TracyImGui.hpp

```diff
@@ -183,8 +183,23 @@ namespace tracy
 
         const auto rw = x1 - x0;
         const auto rh = y1 - y0;
-        const auto v0 = ImVec2( x0, y0 - rw );
         const auto cnt = int( ( rh + rw + sw*2 ) / ( sw*2 ) );
+        auto v0 = ImVec2( x0, y0 - rw );
+
+        if ( fix_stripes_in_screen_space )
+        {
+            const auto window_height = double( ImGui::GetWindowHeight() );
+            const auto flipped_v0y = window_height - v0.y; //we transform into a y-is-up coordinate space to achieve upper-left to lower-right stripes. If we didn't, we would calculate values for lower-left to upper-right
+
+            const auto manhatten_distance = x0 + flipped_v0y;
+            const auto in_multiples_of_2_times_sw = int( manhatten_distance / ( sw*2 ) );
```

Comment:
I'm not sure about this flooring. It will likely break things when `dpiScale` (main.cpp) isn't 1.

Reply:
I'm not understanding how that would happen. This floores a mulitplier, not an actual screen-distance, so I can't see how DPI settings would cause a problem here. Also: I am using two monitors with different DPIs and didn't obsever any problems.

Reply:
Dynamic adapting to DPI changes in runtime is not available in Tracy (it's still ongoing work in ImGui). What you see as adapting is scaling of the window bitmap on the OS level (assuming you're using Windows).

I'm not entirely sure what this variable is used for, so this might be nitpicking. Anyway, this is a close-up of stripes with `dpiScale` set to 1:

![obraz](https://user-images.githubusercontent.com/600573/89639136-54b03f80-d8ad-11ea-983c-c99bec973b10.png)

And this is with `dpiScale` forced to 1.35:

![obraz](https://user-images.githubusercontent.com/600573/89639165-5bd74d80-d8ad-11ea-8086-186b161d0740.png)

There is some misalignment, which _might_ be caused by ignoring the fractional part. I don't know.

Reply:
Wait, but the flooring here is purely to find the place where a stripe should begin. What happens is basically:

`const auto in_multiples_of_2_times_sw = int( manhatten_distance / ( sw*2 ) );`
Calculate how many stripes fit in the manhatten distance. (e.g. 16.32). If this value was fractional, it means that the current manhatten distance is _inside_ of a stripe. We want to place the vertices on the edges of the stripes. We are on an edge if the multiple has no fractional part, so we floor to get to the lower edge of the stripe we are currently in.

`const auto floored_manhatten_distance = double( in_multiples_of_2_times_sw*sw*2 ); //floor in terms of 2 * stripe width`
Then, we multiply by sw*2 again, which gives us the manhattan distance that marks the start of the (in this case) 16th stripe. Note that this value might have a fractional part! If e.g. sw is 2.3, then the lower edge of the 16th stripe will be at 2.3 * 16 = 36.8.

(I just noticed the comment is on the wrong line, it should be on the previous)

And after that, we subtract x from the manhattan distance to get just the y part, and flip the y back to imgui space.

So the flooring is the necessary logical operation of finding the lower edge of the stripe. Without flooring this algorithm would not work.

Reply:
You are of course right that there still is some misalignment, but I am quite sure that the flooring is not the cause.